Tasks/Maven/maven2.js

```diff
@@ -133,6 +132,34 @@ function getEndpointDetails(inputFieldName) {
     };
 }
 
+// The endpoint stores the auth details as JSON. Unfortunately the structure of the JSON has changed through time, namely the keys were sometimes Upercase
+// To work around this, we can perform case insenstive checks in the property dictionary of the object. Note that the PowerShell implementation does not suffer from this problem.
+function getAuthParameter(endpoint, paramName) {
+
+    var paramValue = null;
+    var auth = tl.getEndpointAuthorization(endpoint, false);
+
+    if (auth.scheme != "UsernamePassword") {
+        throw new Error("The authorization scheme " + auth.scheme + " is not supported for a SonarQube endpoint. Please use a username and a password.");
+    }
+
+    var parameters = Object.getOwnPropertyNames(auth['parameters']);
+
+    var keyName;
+    parameters.some(function (key) {
+
+        if (key.toLowerCase() === paramName.toLowerCase()) {
+            keyName = key;
+
+            return true;
+        }
+    });
+
```

Comment:
I'm a bit hesitant to assert because there is no assert in javascript. I could throw instead, however I've logged a bug against the RM team so that they don't make the username / password of a generic endpoint mandatory. This bug has been approved but not fixed. Depending on the fix, it may be possible that the returned JSON will not contain these fields. For more details the bug is https://mseng.visualstudio.com/DefaultCollection/VSOnline/Build/_workitems#_a=edit&newQuery=false&id=415127&triage=true